code/modules/mob/living/carbon/human/species.dm

```diff
@@ -231,164 +255,102 @@ GLOBAL_LIST_EMPTY(features_by_species)
-
-	if(C.get_bodypart(BODY_ZONE_HEAD))
-		if(brain && (replace_current || !should_have_brain))
-			if(!brain.decoy_override)//Just keep it if it's fake
-				brain.Remove(C,TRUE,TRUE)
-				required_organs -= /obj/item/organ/brain
-				QDEL_NULL(brain)
-		if(should_have_brain && !brain)
-			brain = new mutant_brain()
-			brain.Insert(C, TRUE, TRUE)
-			required_organs |= /obj/item/organ/brain
-
-		if(eyes && (replace_current || !should_have_eyes))
-			eyes.Remove(C,1)
-			required_organs -= /obj/item/organ/eyes
-			QDEL_NULL(eyes)
-		if(should_have_eyes && !eyes)
-			eyes = new mutanteyes
-			eyes.Insert(C)
-			required_organs |= /obj/item/organ/eyes
-
-		if(ears && (replace_current || !should_have_ears))
-			ears.Remove(C,1)
-			required_organs -= /obj/item/organ/ears
-			QDEL_NULL(ears)
-		if(should_have_ears && !ears)
-			ears = new mutantears
-			ears.Insert(C)
-			required_organs |= /obj/item/organ/ears
-
-		if(tongue && (replace_current || !should_have_tongue))
-			tongue.Remove(C,1)
-			required_organs -= /obj/item/organ/tongue
-			QDEL_NULL(tongue)
-		if(should_have_tongue && !tongue)
-			tongue = new mutanttongue
-			tongue.Insert(C)
-			required_organs |= /obj/item/organ/tongue
+
+/** regenerate_organs
+  * Corrects organs in a carbon, removing ones it doesn't need and adding ones it does
+  *
+  * takes all organ slots, removes organs a species should not have, adds organs a species should have.
+  * can use replace_current to refresh all organs, creating an entirely new set.
+  * Arguments:
+  * C - carbon, the owner of the species datum AKA whoever we're regenerating organs in
+  * old_species - datum, used when regenerate organs is called in a switching species to remove old mutant organs.
+  * replace_current - boolean, forces all old organs to get deleted whether or not they pass the species' ability to keep that organ
+  * excluded_zones - list, add zone defines to block organs inside of the zones from getting handled. see headless mutation for an example
+  */
+/datum/species/proc/regenerate_organs(mob/living/carbon/C,datum/species/old_species,replace_current=TRUE,list/excluded_zones)
+	//what should be put in if there is no mutantorgan (brains handled seperately)
+	var/list/slot_mutantorgans = list(
+		ORGAN_SLOT_BRAIN = mutantbrain,
+		ORGAN_SLOT_HEART = mutantheart,
+		ORGAN_SLOT_LUNGS = mutantlungs,
+		ORGAN_SLOT_APPENDIX = mutantappendix,
+		ORGAN_SLOT_EYES = mutanteyes,
+		ORGAN_SLOT_EARS = mutantears,
+		ORGAN_SLOT_TONGUE = mutanttongue,
+		ORGAN_SLOT_LIVER = mutantliver,
+		ORGAN_SLOT_STOMACH = mutantstomach,
+		ORGAN_SLOT_WINGS = mutantwings
+	)
+
+	var/list/slot_organs = list(
+		ORGAN_SLOT_BRAIN,
+		ORGAN_SLOT_HEART,
+		ORGAN_SLOT_LUNGS,
+		ORGAN_SLOT_APPENDIX,
+		ORGAN_SLOT_EYES,
+		ORGAN_SLOT_EARS,
+		ORGAN_SLOT_TONGUE,
+		ORGAN_SLOT_LIVER,
+		ORGAN_SLOT_STOMACH,
+		ORGAN_SLOT_WINGS
+	)
+
+	//if theres no added wing type, we want to avoid adding a null
+	if(isnull(mutantwings))
```

Comment:
my glorious hacky wing shitcode that prevents everyone from having wings.

Thanks, 3 yr old zesko pr.